### ISSUE TITLE
size of image is not shown

### ISSUE THREAD

**Issue by mprivoro** [Joxit/docker-registry-ui#296]
![image](https://user-images.githubusercontent.com/7849370/233297087-80345fa7-089d-449c-acdf-3f4d17b81308.png)

cannot see image size for any installation

**Comment by Joxit** [comment]
Hi, thank you for using my project :smile: 

I need more details about your issue :slightly_smiling_face:

**Comment by Aleksey6115** [comment]
https://ibb.co/Zh1n2FZ   or   https://pasteboard.co/LnXxQha0H2BZ.png
https://ibb.co/L5GT1r8   or   https://pasteboard.co/gxA4JjmMnw4T.png
https://ibb.co/n37dcK3   or   https://pasteboard.co/BJlXuGdMI1Ez.png

ununtu 22.04.2
docker 23.0.5
dockercompose 2.17.3
delete not working, push DELETE button doen not gives effect
push working
pull working
history not showing enything
DOCKERFILE button not showing enything, just empty window

**Comment by Joxit** [comment]
I checked your screenshot, please next time fufill correctly a bug report as [defined here](https://github.com/Joxit/docker-registry-ui/issues/new?assignees=&labels=&template=bug_report.md&title=)

So there are no longer an option named `URL` in docker-registry-ui since v2, you may use `REGISTRY_URL` instead.

I would need errors [in your web console](https://balsamiq.com/support/faqs/browserconsole/).

**Comment by Aleksey6115** [comment]
Access to XMLHttpRequest at 'https://10.0.0.224/v2/********integration/manifests/666' from origin 'https://dockerregistry2.domain.tld/' has been blocked by CORS policy: Request header field accept is not allowed by Access-Control-Allow-Headers in preflight response.
docker-registry-ui.js:40          GET https://10.0.0.224/v2/********integration/manifests/666 net::ERR_FAILED


10.0.0.224 is registry container
https://dockerregistry2.domain.tld is Joxit/docker-registry-ui container

With https or without, with ip or with fqdn error the same - has been blocked by CORS policy.

**Comment by Joxit** [comment]
Hi, your error says
> has been blocked by CORS policy: Request header field accept is not allowed by `Access-Control-Allow-Headers` in preflight response.

You missed configured your docker registry, you usually need this configuration
```yaml
http:
  headers:
    Access-Control-Allow-Methods: ['HEAD', 'GET', 'OPTIONS', 'DELETE']
    Access-Control-Allow-Headers: ['Authorization', 'Accept', 'Cache-Control']
    Access-Control-Expose-Headers: ['Docker-Content-Digest']
```

**Comment by xyb** [comment]
I'm experiencing a similar issue as well. It appears that the `/v2/redis/manifests/<tag>` API is not providing the image size and upload time in its response. Here are some additional details::
- Docker Registry UI version: 2.4.1
- Registry version: 2.8.0
- Docker Registry service logs
```log
1.2.3.4 - - [05/Jun/2023:04:41:04 +0000] "GET /v2/redis/tags/list HTTP/1.1" 200 30 "https://registry-ui.my/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/113.0"
time="2023-06-05T04:41:04.38231856Z" level=info msg="response completed" go.version=go1.16.13 http.request.host=registry.my http.request.id=1f0a1236-bdae-4430-96cb-35611df84494 http.request.method=GET http.request.referer="https://registry-ui.my/" http.request.remoteaddr=4.3.2.1 http.request.uri="/v2/redis/tags/list" http.request.useragent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/113.0" http.response.contenttype="application/json; charset=utf-8" http.response.duration=2.487858ms http.response.status=200 http.response.written=30
1.2.3.4 - - [05/Jun/2023:04:41:04 +0000] "OPTIONS /v2/redis/manifests/7 HTTP/1.1" 200 0 "https://registry-ui.my/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/113.0"
time="2023-06-05T04:41:04.433981956Z" level=info msg="response completed" go.version=go1.16.13 http.request.host=registry.my http.request.id=e20161ba-ccd0-43fe-89a6-6bc7fef1e822 http.request.method=OPTIONS http.request.referer="https://registry-ui.my/" http.request.remoteaddr=4.3.2.1 http.request.uri="/v2/redis/manifests/7" http.request.useragent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/113.0" http.response.duration="786.264µs" http.response.status=200 http.response.written=0
```
- Response from the `/v2/redis/manifests/7` API:
```json
{
   "schemaVersion": 1,
   "name": "redis",
   "tag": "7",
   "architecture": "amd64",
   "fsLayers": [
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:db7c27bf355290eeb889d2e2986ba34352eba3adba6ae44eaef63d5e7dd69898"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:ab1a1215d5f9ec95bf6981023be29642373dcdd6218d602d1f049d5e5ee3ee29"
      },
      {
         "blobSum": "sha256:817f7e347ebd5d210d4a5923c221e92275d665ced36e12185bc26747e977e47d"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:51afc2cce3dfa745f19f601b33820e4a960311cd81b0a6a816b1de8e5ae6d7d7"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:ac509f65c3e925781d771dc15fd63c2bd795abdc461f29bd9a3acd0a19f65d3f"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:bb263680fed18eecdc67f885094df6f589bafc19004839d7fdf141df236a61aa"
      }
   ],
   "history": [
      {
         "v1Compatibility": "{\"architecture\":\"amd64\",\"config\":{\"Hostname\":\"\",\"Domainname\":\"\",\"User\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"ExposedPorts\":{\"6379/tcp\":{}},\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":[\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\",\"GOSU_VERSION=1.16\",\"REDIS_VERSION=7.0.8\",\"REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-7.0.8.tar.gz\",\"REDIS_DOWNLOAD_SHA=06a339e491306783dcf55b97f15a5dbcbdc01ccbde6dc23027c475cab735e914\"],\"Cmd\":[\"redis-server\"],\"Image\":\"sha256:9813b5bfa15e896c5ae8153fd2448b839bd886104e3627fec4f892e9248fca6e\",\"Volumes\":{\"/data\":{}},\"WorkingDir\":\"/data\",\"Entrypoint\":[\"docker-entrypoint.sh\"],\"OnBuild\":null,\"Labels\":null},\"container\":\"f77517a83020492f36fe4b41d9a0c6bf9b3154d9fb248131d98b07264f044a03\",\"container_config\":{\"Hostname\":\"f77517a83020\",\"Domainname\":\"\",\"User\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"ExposedPorts\":{\"6379/tcp\":{}},\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":[\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\",\"GOSU_VERSION=1.16\",\"REDIS_VERSION=7.0.8\",\"REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-7.0.8.tar.gz\",\"REDIS_DOWNLOAD_SHA=06a339e491306783dcf55b97f15a5dbcbdc01ccbde6dc23027c475cab735e914\"],\"Cmd\":[\"/bin/sh\",\"-c\",\"#(nop) \",\"CMD [\\\"redis-server\\\"]\"],\"Image\":\"sha256:9813b5bfa15e896c5ae8153fd2448b839bd886104e3627fec4f892e9248fca6e\",\"Volumes\":{\"/data\":{}},\"WorkingDir\":\"/data\",\"Entrypoint\":[\"docker-entrypoint.sh\"],\"OnBuild\":null,\"Labels\":{}},\"created\":\"2023-02-09T06:09:56.878067502Z\",\"docker_version\":\"20.10.12\",\"id\":\"29897524fa45ba3ffc708df48a30ae4d2d5645a74a1b69fdaf08439e6a7bd988\",\"os\":\"linux\",\"parent\":\"a14a12fb986cf96d46d57d75737415ffde8f382c48e6a9de49cae380ed330af3\",\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"a14a12fb986cf96d46d57d75737415ffde8f382c48e6a9de49cae380ed330af3\",\"parent\":\"b63515de456f7ce23e84888af751350fa888764aa154e785db7c5ed5117a1d8d\",\"created\":\"2023-02-09T06:09:56.778219161Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  EXPOSE 6379\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"b63515de456f7ce23e84888af751350fa888764aa154e785db7c5ed5117a1d8d\",\"parent\":\"99e3adf2a5d18c40390df10e0266c4fbdeaaa656a3d52cf13df0cc5fc0806825\",\"created\":\"2023-02-09T06:09:56.677282745Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  ENTRYPOINT [\\\"docker-entrypoint.sh\\\"]\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"99e3adf2a5d18c40390df10e0266c4fbdeaaa656a3d52cf13df0cc5fc0806825\",\"parent\":\"1bdfe20aa1557a8f9dde3390185a1ec4313099db852294b0076d3323522c7541\",\"created\":\"2023-02-09T06:09:56.578982042Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop) COPY file:e873a0e3c13001b5e1c63f9dbe60cc65722af522671787f68b8e8cbcae7d02ac in /usr/local/bin/ \"]}}"
      },
      {
         "v1Compatibility": "{\"id\":\"1bdfe20aa1557a8f9dde3390185a1ec4313099db852294b0076d3323522c7541\",\"parent\":\"dcbe3dd5672f8c67f2a81cc4f4f35981386d9bb581c100a1ecac0c5dfed62dae\",\"created\":\"2023-02-09T06:09:56.462243271Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop) WORKDIR /data\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"dcbe3dd5672f8c67f2a81cc4f4f35981386d9bb581c100a1ecac0c5dfed62dae\",\"parent\":\"2e4d6fd279e57c5407298404106a83d9aca32410dbdc1bce56022167dce4042c\",\"created\":\"2023-02-09T06:09:56.35030732Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  VOLUME [/data]\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"2e4d6fd279e57c5407298404106a83d9aca32410dbdc1bce56022167dce4042c\",\"parent\":\"943e2ed63d758d2664c0322aa071ebabb15f5aac8cd2dc3fd55e61cbb34b066f\",\"created\":\"2023-02-09T06:09:56.246768674Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c mkdir /data \\u0026\\u0026 chown redis:redis /data\"]}}"
      },
      {
         "v1Compatibility": "{\"id\":\"943e2ed63d758d2664c0322aa071ebabb15f5aac8cd2dc3fd55e61cbb34b066f\",\"parent\":\"5e6e51a7d68fbe4d8a03b6d0c82174b6272e171a4f43caceab07ddf6693f7b07\",\"created\":\"2023-02-09T06:09:55.677446152Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c set -eux; \\t\\tsavedAptMark=\\\"$(apt-mark showmanual)\\\"; \\tapt-get update; \\tapt-get install -y --no-install-recommends \\t\\tca-certificates \\t\\twget \\t\\t\\t\\tdpkg-dev \\t\\tgcc \\t\\tlibc6-dev \\t\\tlibssl-dev \\t\\tmake \\t; \\trm -rf /var/lib/apt/lists/*; \\t\\twget -O redis.tar.gz \\\"$REDIS_DOWNLOAD_URL\\\"; \\techo \\\"$REDIS_DOWNLOAD_SHA *redis.tar.gz\\\" | sha256sum -c -; \\tmkdir -p /usr/src/redis; \\ttar -xzf redis.tar.gz -C /usr/src/redis --strip-components=1; \\trm redis.tar.gz; \\t\\tgrep -E '^ *createBoolConfig[(]\\\"protected-mode\\\",.*, *1 *,.*[)],$' /usr/src/redis/src/config.c; \\tsed -ri 's!^( *createBoolConfig[(]\\\"protected-mode\\\",.*, *)1( *,.*[)],)$!\\\\10\\\\2!' /usr/src/redis/src/config.c; \\tgrep -E '^ *createBoolConfig[(]\\\"protected-mode\\\",.*, *0 *,.*[)],$' /usr/src/redis/src/config.c; \\t\\tgnuArch=\\\"$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)\\\"; \\textraJemallocConfigureFlags=\\\"--build=$gnuArch\\\"; \\tdpkgArch=\\\"$(dpkg --print-architecture)\\\"; \\tcase \\\"${dpkgArch##*-}\\\" in \\t\\tamd64 | i386 | x32) extraJemallocConfigureFlags=\\\"$extraJemallocConfigureFlags --with-lg-page=12\\\" ;; \\t\\t*) extraJemallocConfigureFlags=\\\"$extraJemallocConfigureFlags --with-lg-page=16\\\" ;; \\tesac; \\textraJemallocConfigureFlags=\\\"$extraJemallocConfigureFlags --with-lg-hugepage=21\\\"; \\tgrep -F 'cd jemalloc \\u0026\\u0026 ./configure ' /usr/src/redis/deps/Makefile; \\tsed -ri 's!cd jemalloc \\u0026\\u0026 ./configure !\\u0026'\\\"$extraJemallocConfigureFlags\\\"' !' /usr/src/redis/deps/Makefile; \\tgrep -F \\\"cd jemalloc \\u0026\\u0026 ./configure $extraJemallocConfigureFlags \\\" /usr/src/redis/deps/Makefile; \\t\\texport BUILD_TLS=yes; \\tmake -C /usr/src/redis -j \\\"$(nproc)\\\" all; \\tmake -C /usr/src/redis install; \\t\\tserverMd5=\\\"$(md5sum /usr/local/bin/redis-server | cut -d' ' -f1)\\\"; export serverMd5; \\tfind /usr/local/bin/redis* -maxdepth 0 \\t\\t-type f -not -name redis-server \\t\\t-exec sh -eux -c ' \\t\\t\\tmd5=\\\"$(md5sum \\\"$1\\\" | cut -d\\\" \\\" -f1)\\\"; \\t\\t\\ttest \\\"$md5\\\" = \\\"$serverMd5\\\"; \\t\\t' -- '{}' ';' \\t\\t-exec ln -svfT 'redis-server' '{}' ';' \\t; \\t\\trm -r /usr/src/redis; \\t\\tapt-mark auto '.*' \\u003e /dev/null; \\t[ -z \\\"$savedAptMark\\\" ] || apt-mark manual $savedAptMark \\u003e /dev/null; \\tfind /usr/local -type f -executable -exec ldd '{}' ';' \\t\\t| awk '/=\\u003e/ { print $(NF-1) }' \\t\\t| sort -u \\t\\t| xargs -r dpkg-query --search \\t\\t| cut -d: -f1 \\t\\t| sort -u \\t\\t| xargs -r apt-mark manual \\t; \\tapt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \\t\\tredis-cli --version; \\tredis-server --version\"]}}"
      },
      {
         "v1Compatibility": "{\"id\":\"5e6e51a7d68fbe4d8a03b6d0c82174b6272e171a4f43caceab07ddf6693f7b07\",\"parent\":\"c2be39687e29b61054d60e79ece3111f2367a9082a2edd4a88e5318c6affcc5f\",\"created\":\"2023-02-09T06:09:11.196598196Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  ENV REDIS_DOWNLOAD_SHA=06a339e491306783dcf55b97f15a5dbcbdc01ccbde6dc23027c475cab735e914\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"c2be39687e29b61054d60e79ece3111f2367a9082a2edd4a88e5318c6affcc5f\",\"parent\":\"e06e8728fade8553053d9cdebbc8adccac7f608f3ffb0c3f19848742bcca0da4\",\"created\":\"2023-02-09T06:09:11.093050525Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  ENV REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-7.0.8.tar.gz\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"e06e8728fade8553053d9cdebbc8adccac7f608f3ffb0c3f19848742bcca0da4\",\"parent\":\"fbb2a9d350179163e20e6db307d826dc633acf3c1e0e7b182d47147df5c373de\",\"created\":\"2023-02-09T06:09:10.989490413Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  ENV REDIS_VERSION=7.0.8\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"fbb2a9d350179163e20e6db307d826dc633acf3c1e0e7b182d47147df5c373de\",\"parent\":\"88b97167858aa407d28397d3a5c8880fb8f5787e8b70bf6033b766860ed45fca\",\"created\":\"2023-02-09T06:09:10.877003066Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c set -eux; \\tsavedAptMark=\\\"$(apt-mark showmanual)\\\"; \\tapt-get update; \\tapt-get install -y --no-install-recommends ca-certificates dirmngr gnupg wget; \\trm -rf /var/lib/apt/lists/*; \\tdpkgArch=\\\"$(dpkg --print-architecture | awk -F- '{ print $NF }')\\\"; \\twget -O /usr/local/bin/gosu \\\"https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch\\\"; \\twget -O /usr/local/bin/gosu.asc \\\"https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc\\\"; \\texport GNUPGHOME=\\\"$(mktemp -d)\\\"; \\tgpg --batch --keyserver hkps://keys.openpgp.org --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \\tgpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \\tgpgconf --kill all; \\trm -rf \\\"$GNUPGHOME\\\" /usr/local/bin/gosu.asc; \\tapt-mark auto '.*' \\u003e /dev/null; \\t[ -z \\\"$savedAptMark\\\" ] || apt-mark manual $savedAptMark \\u003e /dev/null; \\tapt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \\tchmod +x /usr/local/bin/gosu; \\tgosu --version; \\tgosu nobody true\"]}}"
      },
      {
         "v1Compatibility": "{\"id\":\"88b97167858aa407d28397d3a5c8880fb8f5787e8b70bf6033b766860ed45fca\",\"parent\":\"34f998de5c9847a30040c2bc897148c6501aebd19bd750a564e83429bc99d3fc\",\"created\":\"2023-02-09T06:08:58.86711645Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  ENV GOSU_VERSION=1.16\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"34f998de5c9847a30040c2bc897148c6501aebd19bd750a564e83429bc99d3fc\",\"parent\":\"fadeea921d1edf177afd266f44f4fc89f8a3f141c3e50c1d4322429e7bb9881c\",\"created\":\"2023-02-09T06:08:58.764440818Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c groupadd -r -g 999 redis \\u0026\\u0026 useradd -r -g redis -u 999 redis\"]}}"
      },
      {
         "v1Compatibility": "{\"id\":\"fadeea921d1edf177afd266f44f4fc89f8a3f141c3e50c1d4322429e7bb9881c\",\"parent\":\"0c185fb8df5728945bb955533c532c604ab06da1e97d575a76b79db131db8dd5\",\"created\":\"2023-02-09T03:20:20.798170292Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop)  CMD [\\\"bash\\\"]\"]},\"throwaway\":true}"
      },
      {
         "v1Compatibility": "{\"id\":\"0c185fb8df5728945bb955533c532c604ab06da1e97d575a76b79db131db8dd5\",\"created\":\"2023-02-09T03:20:20.296746756Z\",\"container_config\":{\"Cmd\":[\"/bin/sh -c #(nop) ADD file:3ea7c69e4bfac2ebb6f86baaedab31827c86a594dba8080a49928e211ad3c7a0 in / \"]}}"
      }
   ],
   "signatures": [
      {
         "header": {
            "jwk": {
               "crv": "...",
               "kid": "...",
               "kty": "...",
               "x": "...",
               "y": "..."
            },
            "alg": "ES256"
         },
         "signature": "...",
         "protected": "..."
      }
   ]
}
```

**Comment by Joxit** [comment]
Hello, thank you for using my project. I need more details such as the browser console + network tab

**Comment by colindawson** [comment]
Hi,  I think I'm having the same issue.  let's see if I can help get you the information that you need.
Here's my registry config file.

`version: 0.1
log:
  fields:
    service: registry
storage:
  delete:
    enabled: true
  cache:
    blobdescriptor: inmemory
  filesystem:
    rootdirectory: /var/lib/registry
http:
  addr: :5000
  headers:
    X-Content-Type-Options: [nosniff]
    Access-Control-Allow-Origin: ['http://linux.lan.local:8080']
    Access-Control-Allow-Methods: ['HEAD', 'GET', 'OPTIONS', 'DELETE']
    Access-Control-Allow-Headers: ['Authorization', 'Accept', 'Cache-Control']
    Access-Control-Max-Age: [1728000]
    Access-Control-Allow-Credentials: [true]
    Access-Control-Expose-Headers: ['Docker-Content-Digest']
auth:
  htpasswd:
    realm: basic-realm
    path: /etc/docker/registry/htpasswd`

I'm both containers on the same linux machine using docker-compose.  Here's my yml

`version: '3.1'

services:
  registry-server:
    image: registry:2.8.2
    restart: always
    ports:
      - 5000:5000
    environment:
      REGISTRY_AUTH: htpasswd
      REGISTRY_AUTH_HTPASSWD_PATH: /auth/htpasswd
      REGISTRY_AUTH_HTPASSWD_REALM: Registry Realm
    volumes:
      - ./registry/data:/var/lib/registry
      - ./registry/config.yml:/etc/docker/registry/config.yml:ro
      - ./registry/auth:/auth
    container_name: registry-server

  registry-ui:
    image: joxit/docker-registry-ui:main
    restart: always
    ports:
      - 8080:80
    environment:
      - REGISTRY_URL=http://linux.lan.local:5000
      - DELETE_IMAGES=true
      - SHOW_CONTENT_DIGEST=true
    container_name: registry-ui`


The registry-ui is not on the web, or accessible from outside my private network.   I'm still using authentication, as the registry itself is web published via a reverse proxy.

Here's the console errors
![image](https://github.com/Joxit/docker-registry-ui/assets/5050229/2a708c53-90df-4c4d-bfd6-a1a48272d4e1)

Here's the network tab
![image](https://github.com/Joxit/docker-registry-ui/assets/5050229/343422e2-0d89-4f5f-9afd-6b4e7192a7b7)

**Comment by Joxit** [comment]
Hi @colindawson, your issue is simple, you just need to read the [FAQ](https://github.com/Joxit/docker-registry-ui#faq) and all the linked issues...:
> -   Why OPTIONS (aka preflight requests) and DELETE fails with 401 status code (using Basic Auth) ?
>     -   This is caused by a bug in docker registry, it returns 401 status requests on preflight requests, this breaks [W3C preflight-request specification](https://www.w3.org/TR/cors/#preflight-request). I suggest to have your UI on the same domain than your registry e.g. registry.example.com/ui/ **or** use `NGINX_PROXY_PASS_URL` **or** configure a nginx/apache/haproxy in front of your registry that returns 200 on each OPTIONS requests. (see [#104](https://github.com/Joxit/docker-registry-ui/issues/104), [#204](https://github.com/Joxit/docker-registry-ui/issues/204), [#207](https://github.com/Joxit/docker-registry-ui/issues/207), [#214](https://github.com/Joxit/docker-registry-ui/issues/214), [#266](https://github.com/Joxit/docker-registry-ui/issues/266)).

If you are using `NGINX_PROXY_PASS_URL` solution, don't forget to set `SINGLE_REGISTRY=true` and read carefully the [documentation](https://github.com/Joxit/docker-registry-ui#available-options)

**Comment by colindawson** [comment]
@Joxit I've read through that part of the FAQ, skimmed, the linked issues, looked at the documentation, and it all means absolutely nothing to me.  I'm not using NGINX_PROXY_PASS_URL.   For the purposes of the registry UI, I'm running both containers on the same machine, the registry on port 5000, and the registry-ui on port 8080.

**Comment by Joxit** [comment]
Yeah, but this is exactly what I saw in your screenshot: "preflight request" and "401 status code". Both of them are in the FAQ.

The FAQ gives you three ways to fix your issue: Use the same domain OR use  `NGINX_PROXY_PASS_URL` OR configure a nginx/apache/haproxy in front of your registry. Pick one.

Some working examples from the linked issues that uses `NGINX_PROXY_PASS_URL`: https://github.com/Joxit/docker-registry-ui/issues/204#issuecomment-920267104 and https://github.com/Joxit/docker-registry-ui/issues/204#issuecomment-922348958

Or I can do it for you:

docker-compose.yaml
```yaml
version: '3.1'

services:
  registry-server:
    image: registry:2.8.2
    restart: always
    ports:
      - 5000:5000
    environment:
      REGISTRY_AUTH: htpasswd
      REGISTRY_AUTH_HTPASSWD_PATH: /auth/htpasswd
      REGISTRY_AUTH_HTPASSWD_REALM: Registry Realm
    volumes:
      - ./registry/data:/var/lib/registry
      - ./registry/config.yml:/etc/docker/registry/config.yml:ro
      - ./registry/auth:/auth
    container_name: registry-server

  registry-ui:
    image: joxit/docker-registry-ui:main
    restart: always
    ports:
      - 8080:80
    environment:
      - NGINX_PROXY_PASS_URL=http://registry-server:5000
      - DELETE_IMAGES=true
      - SHOW_CONTENT_DIGEST=true
      - SINGLE_REGISTRY=true
    container_name: registry-ui
```

**Comment by colindawson** [comment]
That fixed it for me.   Everything sounded far too cryptic for me.   Seeing the example above, showed me what to do...

      registry-ui:
        image: joxit/docker-registry-ui:main
        restart: always
        ports:
          - 8080:80
        environment:
          - NGINX_PROXY_PASS_URL=http://registry-server:5000
    #      - REGISTRY_URL=http://linux.lan.local:5000
          - DELETE_IMAGES=true
          - SHOW_CONTENT_DIGEST=true
          - SINGLE_REGISTRY=true
        container_name: registry-ui

In summary the delta change is

remove "- REGISTRY_URL=http://linux.lan.local:5000"
add           - NGINX_PROXY_PASS_URL=http://registry-server:5000
and           - SINGLE_REGISTRY=true

Then hey presto, it's all working as you intended.

![image](https://github.com/Joxit/docker-registry-ui/assets/5050229/a23c6fc4-6cda-443e-b62f-952a32545eef)

Thank you.    It would not surpise me if the OP had the same issue, just didn't know how to articulate the problem.

**Comment by Joxit** [comment]
Nice !
I tried to be as clear as possible in the FAQ :confused: Linking to some example would be better for you ?

**Comment by colindawson** [comment]
> Linking to some example would be better for you ?

Maybe worth adding a docker-compose folder to the examples, and putting in a docker-compose like I have?    That should be enough to get a simple example working.     Also wondering if it's possible to combine the NGINX_PROXY_PASS_URL and REGISTRY_URL parameters, so that there's no need to switch from one to the other.  Again would help to simplify the setup for people who are not familiar with the project.

**Comment by Joxit** [comment]
In fact, they already exists :sweat_smile: Your first [docker-compose.yml](https://github.com/Joxit/docker-registry-ui/blob/main/examples/ui-as-standalone/credentials.yml) and [the new one](https://github.com/Joxit/docker-registry-ui/blob/main/examples/ui-as-proxy/credentials.yml)

The use of `REGISTRY_URL` and `NGINX_PROXY_PASS_URL` is really different.
`REGISTRY_URL` should be use when your registry is on a different URL, but as it says in the README you need CORS configurations

`NGINX_PROXY_PASS_URL` as the name suggest, uses NGINX as a proxy for your registry. This avoid CORS configuration and is simpler for newbies in OPS and web. Maybe I should share this one as default configuration :thinking:

**Comment by colindawson** [comment]
Ah ha.   now I see why I didn't see those files, didn't think to look in that folder.    Here's my logic as to why....

1. Was looking in the examples folder for "docker-compose", didn't find it.
2. Did see "helm" and "kubernetes", so didn't think to keep digging.

Not sure if anything can/should be changed.  Just pointing out what was going on in my head.

Anyways, I love that it's working, and it's giving me much much more information that I expected.   Love the project, it's awesome.

**Comment by Joxit** [comment]
Thank you for you feedback :smile:  I added a recommended docker-compose in the main README with the release of 2.5.0! I think the example part need some rework anyway :smile: 

Thank you, I really appreciate :smiling_face_with_three_hearts:
